endpoints/authentication/src/main.cpp

```diff
@@ -329,17 +329,16 @@ namespace irods::http::handler
 					// Not an error, likely to have id_token
 					// TODO: Consider handling bit flip cases
 					const std::string jwt_token{oidc_response.at("id_token").get_ref<const std::string&>()};
+					auto real_decoded_token{jwt::decode<jwt::traits::nlohmann_json>(jwt_token)};
+					auto decoded_token{
+						openid::validate_using_local_validation(openid::TokenType::ID, real_decoded_token)};
 
-					// Get OIDC token && feed to JWT parser
-					// TODO: Handle case where we throw!!!
-					auto decoded_token{jwt::decode<jwt::traits::nlohmann_json>(jwt_token).get_payload_json()};
-
-					auto irods_username{irods::http::map_json_to_user(decoded_token)};
+					auto irods_username{irods::http::map_json_to_user(*decoded_token)};
```

Comment:
This passes a nullptr to the plugins. Are we fine with this, or do we want to do a optional check before?

Reply:
This also applies to the similar block of code further down below.

Reply:
Passing a nullptr doesn't make much sense to me. Unless there's a really good reason to allow them to be passed to the plugin, I think we check/reject nullptrs.

Reply:
Checked.